include/openthread/srp_server.h

```diff
@@ -304,20 +304,24 @@ uint16_t otSrpServerServiceGetWeight(const otSrpServerService *aService);
 uint16_t otSrpServerServiceGetPriority(const otSrpServerService *aService);
 
 /**
- * This method returns the next TXT entry of the service instance.
+ * This function returns the TXT record data of the service instance.
  *
- * @param[in]     aService   A pointer to the SRP service.
- * @param[inout]  aIterator  A pointer to the TXT iterator context. To get the first
- *                           TXT entry, it should be set to OT_DNS_TXT_ITERATOR_INIT.
- * @param[out]    aTxtEntry  A pointer to where the TXT entry will be placed.
+ * @param[in]  aService  A pointer to the SRP service.
+ *
+ * @returns A pointer to the buffer containing the TXT record data.
+ *
+ */
+const uint8_t *otSrpServerServiceGetTxtData(const otSrpServerService *aService);
+
+/**
+ * This function returns the TXT recored data length of the service instance.
+ *
+ * @param[in]  aService  A pointer to the SRP service.
  *
- * @retval OT_ERROR_NONE       Successfully found the next TXT entry.
- * @retval OT_ERROR_NOT_FOUND  No subsequent TXT entry exists in the service.
+ * @return The TXT record data length (number of bytes in buffer returned from `otSrpServerServiceGetTxtData()`).
  *
  */
-otError otSrpServerServiceGetNextTxtEntry(const otSrpServerService *aService,
-                                          otDnsTxtIterator *        aIterator,
-                                          otDnsTxtEntry *           aTxtEntry);
+uint16_t otSrpServerServiceGetTxtDataLength(const otSrpServerService *aService);
```

Comment:
We originally have a single API that returns the TXT data pointer and TXT data length together
```c++
const uint8_t *otSrpServerServiceGetTxtData(const otSrpServerService *aService, uint16_t *aTxtLength);
```
I think this is better because:
1. There are no cases that we can/should use the TXT data without knowing its length.
2. It reduces one public API.

Thoughts?

Reply:
Using one API SGTM. Will change it to use this model later in next push.